src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -507,13 +507,17 @@ private static short InferTypeArityFromMetadataName(string emittedTypeName, out
 
             // Given a name corresponding to <unmangledName>`<arity>,
             // extract the arity.
-            string stringRepresentingArity = emittedTypeName.Substring(indexOfManglingChar);
+            var aritySpan = emittedTypeName[indexOfManglingChar..];
 
             int arity;
-            bool nonNumericCharFound = !int.TryParse(stringRepresentingArity, NumberStyles.None, CultureInfo.InvariantCulture, out arity);
+#if NET
+            bool nonNumericCharFound = !int.TryParse(aritySpan, NumberStyles.None, CultureInfo.InvariantCulture, out arity);
+#else
+            bool nonNumericCharFound = !int.TryParse(aritySpan.ToString(), NumberStyles.None, CultureInfo.InvariantCulture, out arity);
```

Comment:
only time we have to go back to forcing a string alloc.  only for generic types on netstandard.  :'(

Reply:
💡 It would be super easy to avoid this allocation for single-digit arities, and it would also cover the overwhelming majority of cases.